### PR TITLE
chore: replace `github.com/ghodss/yaml` with `sigs.k8s.io/yaml`

### PR DESCRIPTION
**Pull request by Juneezee**
# Description

The `github.com/ghodss/yaml` package is no longer being actively maintained. `sigs.k8s.io/yaml` is a permanent fork of `github.com/ghodss/yaml`, which is actively maintained by Kubernetes SIG and widely used in K8s projects.

The notable change is that `github.com/ghodss/yaml` uses `gopkg.in/yaml.v2 v2.2.2`, while `sigs.k8s.io/yaml` uses `gopkg.in/yaml.v2 v2.4.0`. You can see the changes between the two versions here: [v2.2.2...v2.4.0](https://github.com/go-yaml/yaml/compare/v2.2.2...v2.4.0), which mostly consists of bug fixes.

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

```
$ make build
rm -f devtron
wire
Warning: /home/jun/Desktop/github/devtron/internal/util/MergeUtil.go:29:6: using struct literal to inject github.com/devtron-labs/devtron/internal/util.MergeUtil is deprecated and will be removed in the next release; use wire.Struct instead, see https://godoc.org/github.com/google/wire#Struct for more information.
wire: github.com/devtron-labs/devtron: wrote /home/jun/Desktop/github/waiting-for-merge/devtron/wire_gen.go
go test ./pkg/pipeline
ok  	github.com/devtron-labs/devtron/pkg/pipeline	0.051s
echo 'test cases ran successfully'
test cases ran successfully
GOOS=linux go build -o devtron \
		-ldflags="-X 'github.com/devtron-labs/devtron/util.GitCommit=ec9f262' \
		-X 'github.com/devtron-labs/devtron/util.BuildTime=2023-05-08T15:55:10Z' \
		-X 'github.com/devtron-labs/devtron/util.ServerMode=FULL'"
```

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [ ] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by Juneezee** [comment]
cc @kartik-579 @pawan-59 @kripanshdevtron @vikramdevtron

**Comment by kartik-579** [comment]
@Juneezee thanks for pointing this out, we will plan this in our upcoming sprints. You can track this ticket for the progress.

**Comment by Juneezee** [comment]
friendly ping @nishant-d @pawan-mehta-dt

**Comment by Juneezee** [comment]
Hi @kartik-579, I just wanted to follow up on the pull request I submitted a while ago. I understand you mentioned that you would review it, and I appreciate your time. If you happen to have some free time now or in the near future, I would be grateful if you could review the PR. Thank you for your attention, and please let me know if there's anything else I can provide or assist with.

File: pkg/appStore/deployment/tool/gitops/AppStoreDeploymentArgoCdService.go [hunk 1]
```diff
@@ -31,9 +35,7 @@ import (
 	"github.com/golang/protobuf/ptypes/timestamp"
 	"go.opentelemetry.io/otel"
 	"go.uber.org/zap"
-	"net/http"
-	"strings"
-	"time"
+	"sigs.k8s.io/yaml"
```
Comment (kartik-579) [inline pkg/appStore/deployment/tool/gitops/AppStoreDeploymentArgoCdService.go:38]: unused import, please remove

**Review by kartik-579 — CHANGES_REQUESTED** [review]

**Comment by kartik-579** [comment]
Hi @Juneezee, sorry for the delay. I have made one comment, please made the required changes and let me know so that i can check running the application once for sanity checks. Also, building the application after the changes is recommended to avoid such cases.

**Comment by kartik-579** [comment]
@Juneezee thanks for the quick changes. The PR looks good to me, will run it through sanity checks and then we can move ahead with the merge. Please allow me some more time.

**Review by kartik-579 — APPROVED** [review]

**Pull request merged**